platform-dom/src/main/java/com/softicar/platform/dom/elements/time/DomTimeInput.java

```diff
@@ -4,88 +4,98 @@
+
+		timeInput.setPlaceholder(placeholder);
 	}
 
 	@Override
 	protected void doSetDisabled(boolean disabled) {
 
-		this.hourInput.setDisabled(disabled);
-		this.minuteInput.setDisabled(disabled);
-		this.secondInput.setDisabled(disabled);
+		this.timeInput.setDisabled(disabled);
 	}
 
 	@Override
 	public Optional<Time> getValue() {
 
+		var tokens = tokenizeInputValue();
 		try {
-			Integer hours = hourInput.getValue().orElse(null);
-			Integer minutes = minuteInput.getValue().orElse(null);
-			Integer seconds = secondInput.getValue().orElse(null);
-
-			if (hours != null && minutes != null && seconds != null) {
-				return Optional.of(new Time(hours, minutes, seconds));
-			} else if (hours == null && minutes == null && seconds == null) {
+			if (tokens.isEmpty() || tokens.stream().allMatch(String::isEmpty)) {
 				return Optional.empty();
+			} else if (validateTokenLength(tokens)) {
+				while (tokens.size() < 3) {
+					tokens.add("0");
+				}
+				var hours = Integer.valueOf(tokens.get(0));
+				var minutes = Integer.valueOf(tokens.get(1));
+				var seconds = Integer.valueOf(tokens.get(2));
+				return Optional.of(new Time(hours, minutes, seconds));
 			} else {
-				throw new IllegalTimeSpecificationException(getValueAsString());
+				throw new IllegalTimeSpecificationException(timeInput.getValueOrNull());
 			}
 		} catch (Exception exception) {
-			throw new IllegalTimeSpecificationException(exception, getValueAsString());
+			throw new IllegalTimeSpecificationException(exception, timeInput.getValueOrNull());
 		}
 	}
 
 	@Override
 	public void setValue(Time time) {
 
 		if (time != null) {
-			hourInput.setValue(time.getHour());
-			minuteInput.setValue(time.getMinute());
-			secondInput.setValue(time.getSecond());
+			timeInput.setValue(time.toString());
 		} else {
-			hourInput.setValue(null);
-			minuteInput.setValue(null);
-			secondInput.setValue(null);
+			timeInput.setValue(null);
 		}
 	}
 
-	private String getValueAsString() {
-
-		return "%s:%s:%s"
-			.formatted(//
-				hourInput.getTextualValue(),
-				minuteInput.getTextualValue(),
-				secondInput.getTextualValue());
-	}
-
-	private DomIntegerInput createInput(IDisplayString title, ITestMarker testMarker) {
+	private DomTextInput createInput(IDisplayString title, ITestMarker testMarker) {
 
-		var input = new DomIntegerInput();
+		var input = new DomTextInput();
 		input.setTitle(title);
 		input.addMarker(testMarker);
 		input.addCssClass(DomCssClasses.DOM_TIME_INPUT_ELEMENT);
 		input.addChangeCallback(DomTimeInput.this::executeChangeCallbacks);
 		return input;
 	}
+
+	private List<String> tokenizeInputValue() {
+
+		List<String> tokens = new ArrayList<>();
+		var inputText = timeInput.getValueOrNull();
+		if (inputText != null && !inputText.isEmpty()) {
+			tokens = new Tokenizer(':', '\\').tokenize(inputText);
+			if (tokens.size() > 3) {
+				throw new IllegalTimeSpecificationException(inputText);
+			}
+		}
+		return tokens;
+	}
```

Comment:
- we should not work with `getValueOrNull()` there is `getValueText()` which never returns `null`
- and we should use `isBlank()`
```suggestion
	private List<String> tokenizeInputValue() {
		if(timeInput.getValueText().isBlank()) {
			return Collections.emptyList();
		} else {
			var tokens = new Tokenizer(':', '\\').tokenize(timeInput.getValueText());
			if (tokens.size() > 3) {
				throw new IllegalTimeSpecificationException(inputText);
			}
			return tokens;
		}
	}
```